src/app/dataset-view/dataset.component.ts

```diff
@@ -282,7 +283,16 @@ export class DatasetComponent extends BaseDatasetDataComponent implements OnInit
 
     public onRunSQLRequest(params: DatasetRequestBySql): void {
         this.datasetService
-            .requestDatasetDataSqlRun(params) // TODO: Propagate limit from UI and display when it was reached
-            .subscribe();
+            // TODO: Propagate limit from UI and display when it was reached
+            .requestDatasetDataSqlRun(params)
+            .pipe(
+                tap(() => {
+                    this.sqlLoading = true;
+                }),
+                delay(100),
+            )
+            .subscribe(() => {
+                this.sqlLoading = false;
+            });
```

Comment:
Use finally. An error may happen with the HTTP request, and interrupt your happy pipe. If error happens, we would not expec the timer to keep ticking.

Reply:
Fixed, use `finalize`